Puller.java

```diff
@@ -17,46 +18,107 @@
-                            //getCellTypeEnum ill be renamed to getCellType starting from version 4.0
-                            if (currentCell.getCellType() == CellType.STRING) {
-                                System.out.print(currentCell.getStringCellValue() + "--");
-                            } else if (currentCell.getCellType() == CellType.NUMERIC) {
-                               System.out.print(currentCell.getNumericCellValue() + "--");
-                            }
-
-                        }
-                        System.out.println();
-
+                    	counter = 0;
+                      currentRow = iterator.next(); // contains employee info and module scores
+                      cellIterator = currentRow.iterator();
+                      
+                      System.out.println("\nrow values:");
+                      Employee newEmp = new Employee();
+                        while (cellIterator.hasNext()) {                           
+                        	
+                        	Cell currentCell = cellIterator.next();  // empid, name, email, classId, m1Score, m2Score
+             
+                        	
+                        	if (counter == 0) {
+                        		newEmp.setEmployeeID(currentCell.getStringCellValue());//Gets Emp ID
+                        		counter++;
+                        	} else if(counter == 1){
+                        		newEmp.setEmployeeName(currentCell.getStringCellValue());//Gets Emp Name
+                        		counter++;
+                        	} else if(counter == 2) {
+                        		newEmp.setEmployeeEmail(currentCell.getStringCellValue());//Gets Emp Email
+                        		counter++;
+                        	} else if(counter == 3) {
+                        		newEmp.setClassID(currentCell.getNumericCellValue()); //Gets Class ID
+                        		counter++;
+                        	}else if (counter >= 4){
+                        		newEmp.addScore(columns.get(counter), currentCell.getNumericCellValue()); //Adds scores to an employee
+                        		counter++;
+                        		
+                        	}else {
+                        		System.out.println("All data entered");
+                        	}
+                        	            
+                        	
+                        	//System.out.println(newEmp.toString());
+                        	
+                        	}
+                        emps.add(newEmp);
                     }
                 } catch (FileNotFoundException e) {
                     e.printStackTrace();
                 } catch (IOException e) {
                     e.printStackTrace();
+                } finally {
+                	
+                	 for(int i = 0; i < emps.size(); i++) {   
+                         System.out.println(emps.get(i) + " ");
+                     } 
+                	workbook.close();
```

Comment:
This should be inside the finally block. Or even better, use a try-with-resources and declare workbook under it

Reply:
It was suppose to be. I'll have to fix that as well. Idk how it got outside